libraries/AP_Servo_Telem/AP_Servo_Telem.cpp

```diff
@@ -152,6 +152,38 @@ void AP_Servo_Telem::write_log()
 }
 #endif  // HAL_LOGGING_ENABLED
 
+// Fill in telem structure if telem is available, return false if not
+bool AP_Servo_Telem::get_telem(const uint8_t servo_index, TelemetryData& telem) const volatile
+{
+    // Check for valid index
+    if (servo_index >= ARRAY_SIZE(_telem_data)) {
+        return false;
+    }
+
+    // Check if data has ever been received for the servo index provided
+    if ((active_mask & (1U << servo_index)) == 0) {
+        return false;
+    }
+
+    const volatile TelemetryData &telem_data = _telem_data[servo_index];
+
+    // Because the structure is volatile we have to copy element wise
```

Comment:
we could just cast to override the volatile?

Reply:
or could memcpy?

Reply:
I couldn't get casting to work. The compiler was too smart. `memcpy` may work....

Reply:
try this: 
```
    telem = *const_cast<TelemetryData*>(&telem_data);
```

Reply:
`memcpy(&telem, (const void*)&telem_data, sizeof(telem));` will work too. But it's not exactly kosher.

Reply:
> try this:
> 
> ```
>     telem = *const_cast<TelemetryData*>(&telem_data);
> ```

I have changed to this. Thanks.